src/query/shard_query_runner.rs

```diff
@@ -136,17 +136,36 @@ impl ShardQueryRunner {
             let d = dir.unwrap();
             let mut path = d.path();
             path.push(".lock");
+            let mut duration_file = d.path();
+            duration_file.push("duration");
             if Path::new(&path).exists() {
                 let created = d.file_name().to_str().unwrap().parse::<u64>().unwrap();
-                if query.start <= created && query.end >= created {
+
+                let duration = if duration_file.exists() {
+                    let mut dur = File::open(duration_file.as_path());
```

Comment:
Throw an error here using `?` and fail the query instead of returning 0

Reply:
The idea is to make the pipeline work even without the duration file - for backward compatibility or else, we will have to start from scratch.